### PR TITLE
community: Add alazy_load to AsyncHtmlLoader

### PR DESCRIPTION
**Pull request by cbornet**
Also fixes a bug that `_scrape` was called and was doing a second HTTP request synchronously.

**Twitter handle:** cbornet_

### CODE REVIEW

File: libs/community/langchain_community/document_loaders/async_html.py [hunk 1]
```diff
@@ -186,21 +171,43 @@ async def fetch_all(self, urls: List[str]) -> Any:
...
+            parser = "xml"
+        else:
+            parser = self.default_parser
+        self._check_parser(parser)
+        soup = BeautifulSoup(text, parser)
+        metadata = _build_metadata(soup, url)
+        return Document(page_content=text, metadata=metadata)
 
+    def lazy_load(self) -> Iterator[Document]:
+        """Lazy load text from the url(s) in web_path."""
```
Comment (cbornet) [inline libs/community/langchain_community/document_loaders/async_html.py:210]: I think it's problematic that load()/lazy_load() uses asyncio.
A thread-pool with a thread-safe synchronous http lib should be used instead.

Comment (eyurtsev) [reply libs/community/langchain_community/document_loaders/async_html.py:210]: Yeah it's forcing users to use nest_asyncio which is not a good solution

**Review by eyurtsev — APPROVED** [review]

**Review by eyurtsev — APPROVED** [review]

**Comment by Nazareka** [comment]
@cbornet After your changes, `_fetch_valid_connection_docs` is no longer used, is that a coincidence?

**Comment by cbornet** [comment]
Indeed. It should probably be removed.

**Comment by Nazareka** [comment]
Actually, I need functionality like that, so it's easier to skip invalid links:
```
except aiohttp.InvalidURL:
    if self.ignore_load_errors:
        logger.error(f"Invalid URL: {url}")
        return ""
```
instead of validating links before passing them to the loader. I will create a PR for this.

**Pull request merged**